pkg/base/replicated.go

```diff
@@ -392,19 +403,29 @@ func findAllKotsHelmCharts(upstreamFiles []upstreamtypes.UpstreamFile, builder t
 			return nil, errors.Wrapf(err, "failed to convert upstream file %s to base", upstreamFile.Path)
 		}
 
-		helmChart, err := ParseHelmChart(baseFile.Content)
-		if err != nil {
-			fmt.Printf("Failed HelmChart contents:\n%s\n", string(baseFile.Content))
-			return nil, errors.Wrapf(err, "failed to parse rendered HelmChart %s", baseFile.Path)
+		switch helmChartVersion {
+		case "v1beta1":
+			helmChart, err := ParseV1Beta1HelmChart(baseFile.Content)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to parse rendered HelmChart %s", baseFile.Path)
+			}
+			kotsV1Beta1HelmCharts = append(kotsV1Beta1HelmCharts, *helmChart)
+		case "v1beta2":
+			helmChart, err := ParseV1Beta2HelmChart(baseFile.Content)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to parse rendered HelmChart %s", baseFile.Path)
+			}
+			kotsV1Beta2HelmCharts = append(kotsV1Beta2HelmCharts, *helmChart)
 		}
-
-		kotsHelmCharts = append(kotsHelmCharts, helmChart)
 	}
 
-	return kotsHelmCharts, nil
+	// filter out any charts where a cooresponding v1beta2 HelmChart exists
+	filteredCharts := kotsutil.FilterV1Beta1ChartsWithV1Beta2Charts(kotsV1Beta1HelmCharts, kotsV1Beta2HelmCharts)
```

Comment:
why add v1beta2 helm charts then filter them? how about just not adding them in the first place?

Reply:
we're filtering the v1beta1 resources if a matching v1beta2 resource exists for the same helm release.  we collect all v1beta2 resources in addition to v1beta1 so that we can make this determination.  should this logic be moved?